code/modules/mob/living/carbon/superior_animal/ai.dm

```diff
@@ -1,9 +1,17 @@
+
+//NOTE: Don't use this proc for finding specific mobs or a very certain object; ultilize GLOBs instead of view()
 /mob/living/carbon/superior_animal/proc/getObjectsInView()
 	objectsInView = objectsInView || view(src, viewRange)
 	return objectsInView
 
+//Use this for all likely human mobs and superior animals
 /mob/living/carbon/superior_animal/proc/getPotentialTargets()
-	return hearers(src, viewRange)
+	var/list/list_to_return = new
+	for(var/atom/thing in (GLOB.mob_list))
```

Comment:
What proof do you have that this is more performant?

I can imagine it'll scale poorly with having a lot of mobs.

Reply:
Hearers is very heavy on the CPU, but i know that big lists can have an impact on memory and may cause OOR so i would like to see some benchmarks of this before approval, and those benchmarks should not include the changes from ` code/modules/mob/living/living_defines.dm` lines 11 and 12.

Reply:
With the new code, the amount of mobs present in each zlevel should be overall less things to look through compared to hearers() which returns a grid of ((viewRange * 2) + 1) x ((viewRange * 2) + 1) around src and all of it's contents.

Reply:
Outdated code, it currently utilizes the following
`for(var/atom/thing in SSmobs.mob_living_by_zlevel[((get_turf(src)).z)])`